src/mcp/client/auth.py

```diff
@@ -291,6 +327,27 @@ async def _register_client(self) -> httpx.Request | None:
 
         registration_data = self.context.client_metadata.model_dump(by_alias=True, mode="json", exclude_none=True)
 
+        # If token_endpoint_auth_method is None, auto-select based on server support
+        if self.context.client_metadata.token_endpoint_auth_method is None:
+            preference_order = ["client_secret_basic", "client_secret_post", "none"]
```

Comment:
I am torn on whether or not we should allow auto-selecting `"none"`. It seems possibly like bad security to allow that, but I suppose if the server allows it then it is ok?

I suppose _ideally_ we should allow the user to pick a list of auth methods they want to allow to be auto-configured, but I am not sure anyone cares enough to want to use it.